iroha_permissions_validators/src/lib.rs

```diff
@@ -158,35 +158,134 @@ pub mod private_blockchain {
 pub mod public_blockchain {
     use super::*;
 
+    /// Origin asset id param used in permission tokens.
+    pub const ASSET_ID_TOKEN_PARAM_NAME: &str = "asset_id";
+    /// Origin asset id param used in permission tokens.
+    pub const ACCOUNT_ID_TOKEN_PARAM_NAME: &str = "account_id";
+    /// Origin asset definition param used in permission tokens.
+    pub const ASSET_DEFINITION_ID_TOKEN_PARAM_NAME: &str = "asset_definition_id";
+
     /// A preconfigured set of permissions for simple use cases.
     pub fn default_permissions() -> PermissionsValidatorBox {
         // Grant instruction checks are or unioned, so that if one permission validator approves this Grant it will succeed.
         let grant_instruction_validator = PermissionsValidatorBuilder::new()
             .with_validator(transfer::GrantMyAssetAccess)
+            .with_validator(unregister::GrantRegisteredByMeAccess)
+            .with_validator(mint::GrantRegisteredByMeAccess)
+            .with_validator(burn::GrantMyAssetAccess)
+            .with_validator(burn::GrantRegisteredByMeAccess)
             .any_should_succeed("Grant instruction validator.");
         PermissionsValidatorBuilder::new()
             .with_recursive_validator(grant_instruction_validator)
-            .with_recursive_validator(transfer::OnlyOwnedAssets.or(transfer::GrantedAssets))
-            .with_recursive_validator(unregister::OnlyAssetsCreatedByThisAccount)
-            .with_recursive_validator(mint::OnlyAssetsCreatedByThisAccount)
-            .with_recursive_validator(burn::OnlyOwnedAssets)
-            .with_recursive_validator(burn::OnlyAssetsCreatedByThisAccount)
+            .with_recursive_validator(transfer::OnlyOwnedAssets.or(transfer::GrantedByAssetOwner))
+            .with_recursive_validator(
+                unregister::OnlyAssetsCreatedByThisAccount.or(unregister::GrantedByAssetCreator),
+            )
+            .with_recursive_validator(
+                mint::OnlyAssetsCreatedByThisAccount.or(mint::GrantedByAssetCreator),
+            )
+            .with_recursive_validator(burn::OnlyOwnedAssets.or(burn::GrantedByAssetOwner))
+            .with_recursive_validator(
+                burn::OnlyAssetsCreatedByThisAccount.or(burn::GrantedByAssetCreator),
+            )
             .with_recursive_validator(keyvalue::AccountSetOnlyForSignerAccount)
```

Comment:
What do you think about increasing readability by introducing macro which will take values and operators (logical ones, like `||` or `&&` or even `=>`)

Reply:
I actually don't think that using operands will improve readability. Using macro to my mind is not needed here.